### PR TITLE
[Excel] (Custom Functions) Adding volatile functions info back in

### PR DESCRIPTION
**Pull request by mscharlock**

### CODE REVIEW

**Comment by o365devx** [comment]
OPS Build status updates of commit _[256a5fe](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/256a5fe95cc2993c04b7de27c7d833c154014f75)_: 

:clock10: **Preparing**: average preparing time is _44 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[256a5fe](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/256a5fe95cc2993c04b7de27c7d833c154014f75)_: 

:clock10: **Incremental building**: average incremental building time is _29 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[256a5fe](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/256a5fe95cc2993c04b7de27c7d833c154014f75)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-vols/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C2%5C7%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201902070028478197-738%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=lKNkoL0vBjV0HdjSdR1OTCADNMraujToma1r4iObWpI%3D&st=2019-02-07T00%3A25%3A04Z&se=2019-03-10T00%3A30%3A04Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/custom-functions-overview.md [hunk 1]
```diff
@@ -283,6 +284,31 @@ In some situations, you may need to cancel the execution of a streaming custom f
...
+
+Excel contains several built-in volatile functions, such as `RAND` and `TODAY`. For a comprehensive list of Excel’s volatile functions, see [Volatile and Non-Volatile Functions](https://docs.microsoft.com/en-us/office/client-developer/excel/excel-recalculation#volatile-and-non-volatile-functions).
+
+Custom functions allow you to create your own volatile functions, which may be useful when handling dates, times, random numbers, and modelling. For example, Monte Carlo simulations require generation of random inputs to determine an optimal solution.
+
+To declare a function volatile, add `"volatile": true` within the `options` object  for the function in the JSON metadata file, as shown in the following code sample. Note that a function cannot be marked both `"streaming": true` and `"volatile": true`; in the case where both are marked `true` the volatile option will be ignored.
+
+```json
+{
+  "name": "TOMORROW",
```
Comment (davidchesnut) [inline docs/excel/custom-functions-overview.md]:
```suggestion
 "id": "TOMORROW",
  "name": "TOMORROW",
```

Comment (davidchesnut) [reply docs/excel/custom-functions-overview.md]: Need id property to get it to work.

File: docs/excel/custom-functions-overview.md [hunk 2]
```diff
@@ -123,7 +123,8 @@ The following code in **customfunctions.json** specifies the metadata for the `a
     ],
     "options": {
         "cancelable": true,
-        "stream": true
+        "stream": true,
+         "volatile": false
```
Comment (davidchesnut) [inline docs/excel/custom-functions-overview.md]: @mscharlock I think there was a change and you don't actually have to specify the volatile property. I get a warning when I try to use it that my JSON is not valid. If I remove it completely my volatile function works fine. Double-check but I don't think you need to change your JSON.

File: docs/excel/custom-functions-overview.md [hunk 3]
```diff
@@ -140,7 +141,7 @@ The following table lists the properties that are typically present in the JSON
 | `description`	| Describes what the function does. This value appears as a tooltip when the function is the selected item in the autocomplete menu within Excel. |
 | `result` 	| Object that defines the type of information that is returned by the function. For detailed information about this object, see [result](custom-functions-json.md#result). |
 | `parameters` | Array that defines the input parameters for the function. For detailed information about this object, see [parameters](custom-functions-json.md#parameters). |
-| `options`	| Enables you to customize some aspects of how and when Excel executes the function. For more information about how this property can be used, see [Streaming functions](#streaming-functions) and [Canceling a function](#canceling-a-function). |
+| `options`	| Enables you to customize some aspects of how and when Excel executes the function. For more information about how this property can be used, see [Streaming functions](#streaming-functions), [Canceling a function](#canceling-a-function), and [Declaring a volatile function](#declaring-a-volatile-function) later in this article. |
```
Comment (davidchesnut) [inline docs/excel/custom-functions-overview.md]: double-check as it doesn't look like you need to specify an option to have a volatile function.

File: docs/excel/custom-functions-overview.md [hunk 4]
```diff
@@ -283,6 +284,31 @@ In some situations, you may need to cancel the execution of a streaming custom f
...
+
+Excel contains several built-in volatile functions, such as `RAND` and `TODAY`. For a comprehensive list of Excel’s volatile functions, see [Volatile and Non-Volatile Functions](https://docs.microsoft.com/en-us/office/client-developer/excel/excel-recalculation#volatile-and-non-volatile-functions).
+
+Custom functions allow you to create your own volatile functions, which may be useful when handling dates, times, random numbers, and modelling. For example, Monte Carlo simulations require generation of random inputs to determine an optimal solution.
+
+To declare a function volatile, add `"volatile": true` within the `options` object  for the function in the JSON metadata file, as shown in the following code sample. Note that a function cannot be marked both `"streaming": true` and `"volatile": true`; in the case where both are marked `true` the volatile option will be ignored.
```
Comment (davidchesnut) [inline docs/excel/custom-functions-overview.md]: Looks like you just return any value you want. No need to mark the function in any way. But double-check.

**Comment by o365devx** [comment]
OPS Build status updates of commit _[592f57d](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/592f57d134df1a16162621776622596b42ebe8ec)_: 

:clock10: **Preparing**: average preparing time is _44 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[592f57d](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/592f57d134df1a16162621776622596b42ebe8ec)_: 

:clock10: **Incremental building**: average incremental building time is _29 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[592f57d](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/592f57d134df1a16162621776622596b42ebe8ec)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-vols/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C2%5C9%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201902090038220643-738%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=R0FtEUt9afIOdE2UhIoqQybf7fAZDNVpdTsNrt2y3bs%3D&st=2019-02-09T00%3A34%3A40Z&se=2019-03-12T00%3A39%3A40Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
OPS Build status updates of commit _[adcd43e](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/adcd43e360e8fbafab13ae84ea8b795f96b0f154)_: 

:clock10: **Preparing**: average preparing time is _44 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[adcd43e](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/adcd43e360e8fbafab13ae84ea8b795f96b0f154)_: 

:clock10: **Incremental building**: average incremental building time is _29 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[adcd43e](https://github.com/OfficeDev/office-js-docs-pr/pull/738/commits/adcd43e360e8fbafab13ae84ea8b795f96b0f154)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-vols/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C2%5C9%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201902090043550232-738%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=lCHgMyiINvLfovCT6p9HHyjgcTrxaN3Lvl8AnNWquw0%3D&st=2019-02-09T00%3A40%3A09Z&se=2019-03-12T00%3A45%3A09Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by keyur32** [comment]
> Just check with Keyur and folks if you have to specify volatile in the JSON. I don't think you do.
> 
> Cheers,
> David

It is required still to specify that option in the JSON.  thanks.

(I think there are other things we no longer require, i.e. return type)

**Pull request merged**